identity-account/src/account/builder.rs

```diff
@@ -101,39 +80,28 @@ where
 
   /// Sets the account storage adapter.
   #[must_use]
-  pub fn storage(mut self, value: AccountStorage) -> Self {
-    self.storage_template = Some(value);
+  pub fn storage<S: Storage + 'static>(mut self, value: S) -> Self {
+    self.storage = Some(Arc::new(value));
     self
   }
 
-  async fn get_storage(&mut self) -> Result<Arc<dyn Storage>> {
-    match self.storage_template.take() {
-      Some(AccountStorage::Memory) => {
-        let storage = Arc::new(MemStore::new());
-        self.storage = Some(storage);
-      }
-      #[cfg(feature = "stronghold")]
-      Some(AccountStorage::Stronghold(snapshot, password, dropsave)) => {
-        let passref: Option<&str> = password.as_deref();
-        let adapter: Stronghold = Stronghold::new(&snapshot, passref, dropsave).await?;
-
-        if let Some(mut password) = password {
-          password.zeroize();
-        }
-
-        let storage = Arc::new(adapter);
-        self.storage = Some(storage);
-      }
-      Some(AccountStorage::Custom(storage)) => {
-        self.storage = Some(storage);
-      }
-      None => (),
-    };
-
-    // unwrap is fine, since by default, storage_template is `Some`,
-    // which results in storage being `Some`.
-    // Overwriting storage_template always produces `Some` storage.
-    Ok(Arc::clone(self.storage.as_ref().unwrap()))
+  /// Sets the account storage adapter from a shared pointer.
+  #[must_use]
+  pub fn storage_shared(mut self, value: Arc<dyn Storage>) -> Self {
+    self.storage = Some(value);
+    self
+  }
+
+  fn get_storage(&mut self) -> Arc<dyn Storage> {
+    if let Some(storage) = self.storage.as_ref() {
+      Arc::clone(storage)
+    } else {
+      // TODO: throw an error or log a warning that MemStore is not persistent and should not
+      //       be used in production?
```

Comment:
Still undecided whether we should make `storage` mandatory at this point. Requiring passing `MemStore` also means exposing it to the Wasm bindings explicitly or using the custom Typescript `MemStore` implementation, which I believe still needs some changes in #719.

We have mandatory fields in quite a few builders that cause runtime errors if omitted, so I don't think it would be necessary to add `storage` as a constructor parameter to `AccountBuilder` if we choose to make it mandatory.

> ... unconventional as builders go, since it is not consumed when creating an `Account` from it.
I think non-consumption is considered an advantage to be fair.

Reply:
The Wasm bindings are a good point. In light of that, I'm happy with leaving it as-is currently, so we don't have to explicitly expose the `MemStore` now.